tests/logprob/test_order.py

```diff
@@ -43,8 +51,63 @@
 def test_max():
     x = pt.random.normal(0, 1, size=(3,))
     x_name = "x"
-    x_max = pt.max_and_argmax(x, axis=-1)
+    x_max = pt.max(x, axis=-1)
     x_max_value = pt.vector("x_max_value")
-    x_max_logprob = logp(x_max[0], x_max_value)
+    x_max_logprob = logp(x_max, x_max_value)
 
     assert_no_rvs(x_max_logprob)
+
+
+def test_argmax():
+    x = pt.random.normal(0, 1, size=(3,))
+    x_name = "x"
+    x_max = pt.argmax(x, axis=-1)
+    x_max_value = pt.vector("x_max_value")
+
+    with pytest.raises(RuntimeError, match=re.escape("Logprob method not implemented for Argmax")):
+        x_max_logprob = logp(x_max, x_max_value)
+
+
+def test_max_non_iid_fails():
+    x = pm.Normal.dist([0, 1, 2, 3, 4], 1, shape=(5,))
+    x_name = "x"
+    x_max = pt.max(x, axis=-1)
+    x_max_value = pt.vector("x_max_value")
+    with pytest.raises(RuntimeError, match=re.escape("Logprob method not implemented")):
+        x_max_logprob = logp(x_max, x_max_value)
+
+
+# def test_max_non_rv_fails():
+#     x = pt.exp(pm.Beta.dist(1, 1))
+#     x_name = "x"
+#     x_max = pt.max(x, axis=-1)
+#     x_max_value = pt.vector("x_max_value")
+#     # with pytest.raises(RuntimeError, match=re.escape("Logprob method not implemented")):
+#     x_max_logprob = logp(x_max, x_max_value)
+
+
+def test_max_categorical():
+    x = pm.Categorical.dist([1, 1, 1, 1], shape=(5,))
+    x_name = "x"
+    x_max = pt.max(x, axis=-1)
+    x_max_value = pt.vector("x_max_value")
+    # REASON for this failing is lines 71 - 74 and not the expected 89-91 which would pass?
+    with pytest.raises(RuntimeError, match=re.escape("Logprob method not implemented")):
+        x_max_logprob = logp(x_max, x_max_value)
+
+
+def test_max_logprob():
```

Comment:
With your current setup, we're looking for something along the lines of:

```python
test_value = 0.8 # or a vector of your choice

beta_rv = pt.random.beta(3, 1, name="beta")
beta_vv = beta_rv.clone()
beta_rv_logprob = logp(beta_rv, beta_vv)

assert np.isclose(
    beta_rv_logprob.eval({beta_vv: test_value}),
    x_max_logprob.eval({x_max_value: test_value})
)
```
Mathematically, the pdf of an order statistics from uniform i.i.d. RVs is equivalent to a Beta distribution, with alpha = k and beta = n - k + 1 (respectively 3 and 1 in the max with n = 3).

The code above may not be 100% right, but I think that the idea is there: the logprob of whatever test value you use should be the same whether you directly use a beta log-pdf or via your rewrite.

After this, you can perhaps extend this test to testing out multiple value of `test_value`, different values of the size argument, which you can do nicely via `pytest.mark.parametrize`, etc.

Reply:
In short, this is a nice test because, with uniform i.i.d. RVs, you can verify your derivation of the log-probability with a nice mathematical property.